### PR TITLE
json: fix URIs handling bug and add associated unit tests

### PR DESCRIPTION
**Pull request by glemercier**
This PR adds two new unit tests for JSON parsing, one of which causes a bug in parsing URIs with a resource ID included. It also fixes the issue in the JSON parsing code.

### CODE REVIEW

File: tests/tlv_json_lwm2m_data_test.c [hunk 1]
```diff
@@ -384,6 +384,26 @@ static void test_10(void)
...
+}
+
+static void test_12(void)
+{
+    const char * buffer = "{\"bn\":\"/5/0\",                \
+                         \"e\":[                            \
+                           {\"n\":\"/1\",\"sv\":\"http\"}]  \
+                      }";
+
+    test_raw(NULL, (uint8_t *)buffer, strlen(buffer), LWM2M_CONTENT_JSON, "11");
```
Comment (sbertin-telular) [inline tests/tlv_json_lwm2m_data_test.c]: I think the last argument here should be "12".

Comment (glemercier) [reply tests/tlv_json_lwm2m_data_test.c]: that's correct, let me make the change

**Comment by dnav** [comment]
Hi,

Thanks for this.

For being able to merge your PR you need to update your Eclipse CLA to match your commit email address. See: https://wiki.eclipse.org/Development_Resources/Contributing_via_Git#Eclipse_Foundation_Contributor_License_Agreement

Regards,

**Comment by dnav** [comment]
Redone in #392.

**Pull request closed without merging**